### PR TITLE
CRIMAPP-732 Fix task list for capital, routing for shopping baskets, store has no values

### PR DESCRIPTION
**Pull request by timpeat**
## Link to relevant ticket
[CRIMAPP-732](https://dsdmoj.atlassian.net/browse/CRIMAPP-732)

## How to manually test the feature

[CRIMAPP-732]: https://dsdmoj.atlassian.net/browse/CRIMAPP-732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/presenters/tasks/capital_assessment.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Tasks
+  class CapitalAssessment < BaseTask
+    include TypeOfMeansAssessment
+
+    def path
+      edit_steps_capital_property_type_path
+    end
```
Comment (hiboabd) [inline app/presenters/tasks/capital_assessment.rb:7]: Only certain application types go through the full capital assessment so depending on the case type, they may need to be routed to the [trust fund page](https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/blob/main/app/services/decisions/outgoings_decision_tree.rb#L35)

File: app/validators/income_benefits_validator.rb [hunk 2]
```diff
@@ -5,23 +5,19 @@ def validate(record)
...
     record.types.each_with_index do |type, index|
-      next if type == IncomeBenefitType::NONE.to_s
+      next if type == 'none'
 
       income_benefit = record.public_send(type)
       add_indexed_errors(income_benefit, index) unless income_benefit.valid?
     end
 
-    require_type?
+    return unless record.types.empty?
```
Comment (hiboabd) [inline app/validators/income_benefits_validator.rb:14]: Would this be better to go before the loop?

File: app/views/steps/capital/answers/edit.html.erb [hunk 3]
```diff
@@ -7,11 +7,15 @@
...
-            <%= e.message %>
-          </h2>
-        <% end %>
+        <h2 class="govuk-error-summary__title">
+          <%= t(:incomplete_records_heading, scope: [:errors, :error_summary]) %>
+        </h2>
+
+        <ul class="govuk-list govuk-error-summary__list">
+          <% @form_object.capital.errors.where(:base).each do |e| %>
+            <li><%= link_to e.message, edit_steps_capital_property_type_path %></li>
```
Comment (hiboabd) [inline app/views/steps/capital/answers/edit.html.erb:16]: Same here, the capital routing differs depending on the case type

File: config/locales/en/errors.yml [hunk 4]
```diff
@@ -6,6 +6,7 @@ en:
 
     error_summary:
       heading: There is a problem on this page
+      incomplete_records_heading: There is a problem
```
Comment (hiboabd) [inline config/locales/en/errors.yml:9]: Would be good to get a demo of this, can't figure out how to get this to appear

File: app/validators/income_benefits_validator.rb (hunk 2 above)
Comment (timpeat) [reply app/validators/income_benefits_validator.rb:14]: I have fixed this in the task list.

**Review by arthurashman — APPROVED** [review]

**Review by hiboabd — APPROVED** [review]

**Pull request merged**